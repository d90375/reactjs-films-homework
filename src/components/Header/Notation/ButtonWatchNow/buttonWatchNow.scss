.btn__watch {
  background-color: #0aaee4;
  border: 1px solid #0aaee4;
  border-radius: 5px;

  &:hover {
    background: lighten(#0aaee4, 10%);
  }
}
