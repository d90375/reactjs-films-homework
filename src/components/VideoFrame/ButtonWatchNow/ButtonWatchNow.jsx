import PropTypes from "prop-types";
import React from "react";
import styles from "./buttonWatchNow.scss";

const ButtonWatchNow = ({ handleShowTrailer }) => {
  return (
    <>
      <button
        onClick={handleShowTrailer}
        type="button"
        data-id="button-watch-now"
        className={`${styles.btn} ${styles.watchBtn}`}
      >
        Watch Now
      </button>
    </>
  );
};

export default ButtonWatchNow;

ButtonWatchNow.propTypes = {
  handleShowTrailer: PropTypes.func.isRequired
};
