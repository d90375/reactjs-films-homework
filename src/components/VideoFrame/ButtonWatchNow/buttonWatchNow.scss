.watchBtn {
  background-color: #0aaee4;
  border: 1px solid #0aaee4;
  border-radius: 5px;

  &:hover {
    background: lighten(#0aaee4, 10%);
  }
}

.btn {
  composes: btn from '../../../components/Header/Notation/notation.scss';
}
