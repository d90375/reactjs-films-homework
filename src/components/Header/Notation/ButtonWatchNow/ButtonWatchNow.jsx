import PropTypes from "prop-types";
import React from "react";
import { useDispatch } from "react-redux";
import { fetchTrailerById } from "../../../../modules/trailerData/trailerActions";
import "./buttonWatchNow.scss";

const ButtonWatchNow = ({ movieId }) => {
  const dispatch = useDispatch();

  const onShowTrailer = () => {
    dispatch(fetchTrailerById(movieId));
  };

  return (
    <>
      <button onClick={onShowTrailer} type="button" className="btn btn__watch">
        Watch Now
      </button>
    </>
  );
};

export default ButtonWatchNow;

ButtonWatchNow.propTypes = {
  movieId: PropTypes.number.isRequired
};
