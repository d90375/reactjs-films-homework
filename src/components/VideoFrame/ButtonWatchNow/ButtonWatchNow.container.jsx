import PropTypes from "prop-types";
import React, { useCallback } from "react";
import { useDispatch } from "react-redux";

import ButtonWatchNow from "./ButtonWatchNow";
import { fetchTrailerById } from "../../../modules/trailerData";

const ButtonWatchNowContainer = ({ movieId }) => {
  const dispatch = useDispatch();

  const handleShowTrailer = useCallback(() => dispatch(fetchTrailerById(movieId)), [movieId, dispatch]);
  return (
    <>
      <ButtonWatchNow handleShowTrailer={handleShowTrailer} />
    </>
  );
};

export default React.memo(ButtonWatchNowContainer);

ButtonWatchNow.propTypes = {
  movieId: PropTypes.number
};

ButtonWatchNow.defaultProps = {
  movieId: 0
};

ButtonWatchNow.displayName = "ButtonWatchNow";
