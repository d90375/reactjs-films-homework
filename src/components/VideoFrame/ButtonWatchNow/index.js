import ButtonWatchNow from "./ButtonWatchNow.container";

export default ButtonWatchNow;
