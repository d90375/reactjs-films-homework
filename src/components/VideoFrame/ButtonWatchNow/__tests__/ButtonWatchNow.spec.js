import React from "react";
import { shallow } from "enzyme";
import ButtonWatchNow from "../ButtonWatchNow";

describe("ButtonWatchNow", () => {
  let component;

  const mockFunction = jest.fn();

  it("should render ButtonWatchNow component with props", () => {
    component = shallow(<ButtonWatchNow handleShowTrailer={mockFunction} />);
    expect(component).toMatchSnapshot();
  });

  it("should handle show trailer", () => {
    expect(mockFunction.mock.calls.length).toBe(0);
    component.find("[data-id='button-watch-now']").simulate("click");
    expect(mockFunction.mock.calls.length).toBe(1);
  });
});
