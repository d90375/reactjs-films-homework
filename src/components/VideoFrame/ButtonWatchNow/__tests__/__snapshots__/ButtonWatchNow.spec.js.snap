// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWatchNow should render ButtonWatchNow component with props 1`] = `
<Fragment>
  <button
    className="btn watchBtn"
    data-id="button-watch-now"
    onClick={[MockFunction]}
    type="button"
  >
    Watch Now
  </button>
</Fragment>
`;
